arch/rv32i/src/pmp.rs

```diff
@@ -478,148 +492,161 @@ impl kernel::mpu::MPU for PMPConfig {
-                                    + csr::pmpconfig::pmpcfg::a0::TOR,
-                            );
-                            csr::CSR.pmpaddr8.set((start as u32) >> 2);
-
-                            // Set access to end address
-                            csr::CSR.pmpcfg2.set(cfg_val << 8 | csr::CSR.pmpcfg0.get());
-                            csr::CSR.pmpaddr9.set((start as u32 + size as u32) >> 2);
-                        }
-                        5 => {
-                            // Disable access up to the start address
-                            csr::CSR.pmpcfg2.modify(
-                                csr::pmpconfig::pmpcfg::r3::CLEAR
-                                    + csr::pmpconfig::pmpcfg::w3::CLEAR
-                                    + csr::pmpconfig::pmpcfg::x3::CLEAR
-                                    + csr::pmpconfig::pmpcfg::a3::TOR,
-                            );
-                            csr::CSR.pmpaddr10.set((start as u32) >> 2);
-
-                            // Set access to end address
-                            csr::CSR.pmpcfg2.set(cfg_val << 24 | csr::CSR.pmpcfg0.get());
-                            csr::CSR.pmpaddr11.set((start as u32 + size as u32) >> 2);
-                        }
-                        6 => {
-                            // Disable access up to the start address
-                            csr::CSR.pmpcfg3.modify(
-                                csr::pmpconfig::pmpcfg::r0::CLEAR
-                                    + csr::pmpconfig::pmpcfg::w0::CLEAR
-                                    + csr::pmpconfig::pmpcfg::x0::CLEAR
-                                    + csr::pmpconfig::pmpcfg::a0::TOR,
-                            );
-                            csr::CSR.pmpaddr12.set((start as u32) >> 2);
-
-                            // Set access to end address
-                            csr::CSR.pmpcfg3.set(cfg_val << 8 | csr::CSR.pmpcfg0.get());
-                            csr::CSR.pmpaddr13.set((start as u32 + size as u32) >> 2);
-                        }
-                        7 => {
-                            // Disable access up to the start address
-                            csr::CSR.pmpcfg3.modify(
-                                csr::pmpconfig::pmpcfg::r3::CLEAR
-                                    + csr::pmpconfig::pmpcfg::w3::CLEAR
-                                    + csr::pmpconfig::pmpcfg::x3::CLEAR
-                                    + csr::pmpconfig::pmpcfg::a3::TOR,
-                            );
-                            csr::CSR.pmpaddr14.set((start as u32) >> 2);
-
-                            // Set access to end address
-                            csr::CSR.pmpcfg3.set(cfg_val << 24 | csr::CSR.pmpcfg0.get());
-                            csr::CSR.pmpaddr15.set((start as u32 + size as u32) >> 2);
+    fn configure_mpu(&self, config: &Self::MpuConfig, app_id: &AppId) {
+        self.last_configured_for.map(|last_app_id| {
```

Comment:
I think it could be useful to document what this block does.

Reply:
Latest version should be clearer but added a short comment nevertheless.